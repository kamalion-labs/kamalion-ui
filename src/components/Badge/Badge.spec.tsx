import React from 'react';
import renderer from 'react-test-renderer';
import { Badge } from '.';

describe('Badge', () => {
  it("should return null if there's no children", () => {
    const alert = renderer.create(<Badge type="success">{null}</Badge>).toJSON();

    expect(alert).toEqual(null);
  });

  it('should render danger Badge', () => {
    const badge = renderer.create(<Badge type="danger">Danger</Badge>).toJSON();

    expect(badge).toMatchSnapshot();
  });

  it('should render warning Badge', () => {
    const badge = renderer.create(<Badge type="warn">Warning</Badge>).toJSON();

    expect(badge).toMatchSnapshot();
  });

  it('should render info Badge', () => {
    const badge = renderer.create(<Badge type="info">Info</Badge>).toJSON();

    expect(badge).toMatchSnapshot();
  });

  it('should render success Badge', () => {
    const badge = renderer.create(<Badge type="success">Success</Badge>).toJSON();

    expect(badge).toMatchSnapshot();
  });
});
