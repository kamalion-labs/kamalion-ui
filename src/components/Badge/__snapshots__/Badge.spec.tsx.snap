// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render danger Badge 1`] = `
<div
  className="badge badge-danger undefined"
>
  Danger
</div>
`;

exports[`Badge should render info Badge 1`] = `
<div
  className="badge badge-info undefined"
>
  Info
</div>
`;

exports[`Badge should render success Badge 1`] = `
<div
  className="badge badge-success undefined"
>
  Success
</div>
`;

exports[`Badge should render warning Badge 1`] = `
<div
  className="badge badge-warn undefined"
>
  Warning
</div>
`;
